nfts/src/tests/extrinsics.rs

```diff
@@ -9,375 +9,548 @@ fn origin(account: u64) -> mock::Origin {
 	RawOrigin::Signed(account).into()
 }
 
-/* fn root() -> mock::Origin {
+fn root() -> mock::Origin {
 	RawOrigin::Root.into()
-} */
+}
 
-mod transfer {
+mod create {
 	use super::*;
 
 	#[test]
-	fn cannot_transfer_delegated_nfts() {
-		ExtBuilder::new_build(vec![(ALICE, 100)]).execute_with(|| {
-			let nft_id = NFTs::create_nft(ALICE, vec![0], None).unwrap();
-			assert_ok!(NFTs::set_viewer(nft_id, Some(BOB)));
+	fn create() {
+		ExtBuilder::new_build(vec![(ALICE, 1000)]).execute_with(|| {
+			let alice: mock::Origin = origin(ALICE);
+			let data = NFTData::new_default(ALICE, vec![1], vec![50]);
+			let alice_balance = Balances::free_balance(ALICE);
 
-			let ok = NFTs::transfer(origin(ALICE), nft_id, BOB);
-			assert_noop!(ok, Error::<Test>::CannotTransferDelegatedNFTs);
+			// Create NFT with new serie id while there is no series already registered
+			let ok = NFTs::create(
+				alice.clone(),
+				data.ipfs_reference.clone(),
+				Some(data.series_id.clone()),
+			);
+			assert_ok!(ok);
+			let nft_id = NFTs::nft_id_generator() - 1;
+
+			// Final state checks
+			assert_eq!(NFTs::series_id_generator(), 0);
+			let nft_series_details = Some(NFTSeriesDetails::new(ALICE, true));
+			assert_eq!(NFTs::series(&data.series_id), nft_series_details);
+			assert_eq!(NFTs::data(0), Some(data.clone()));
+			assert_eq!(Balances::free_balance(ALICE), alice_balance - NFTs::nft_mint_fee());
+
+			// Events checks
+			let event = NFTsEvent::NFTCreated {
+				nft_id,
+				owner: data.owner,
+				series_id: data.series_id,
+				ipfs_reference: data.ipfs_reference,
+				mint_fee: NFTs::nft_mint_fee(),
+			};
+			let event = Event::NFTs(event);
+			assert_eq!(System::events().last().unwrap().event, event);
 		})
 	}
-}
 
-mod burn {
-	use super::*;
+	#[test]
+	fn create_without_series() {
+		ExtBuilder::new_build(vec![(ALICE, 1000)]).execute_with(|| {
+			let owner = ALICE;
+			let ipfs_reference = vec![1];
+			let alice_balance = Balances::free_balance(ALICE);
+
+			// Create NFT with new serie id while there is no series already registered
+			let ok = NFTs::create(origin(ALICE), ipfs_reference.clone(), None);
+			assert_ok!(ok);
+			let nft_id = NFTs::nft_id_generator() - 1;
+
+			// Final state checks
+			let data = NFTs::data(0);
```

Comment:
Magic number 0

Reply:
Just call unwrap here and call it a day.

Reply:
Actually. Call assert!(data.is_some())
And then resign data = data.unwrap();

Reply:
You know, just call unwrap and that's all

Reply:
I used nft_id, and is_some(), Done